src/vs/workbench/api/common/extHostLanguageFeatures.ts

```diff
@@ -1033,12 +1033,19 @@ class CompletionsAdapter {
 			return undefined;
 		}
 
-		return this._convertCompletionItem({
+		const enforcedResolvedItem = {
 			...item,
 			documentation: resolvedItem.documentation,
 			detail: resolvedItem.detail,
 			additionalTextEdits: resolvedItem.additionalTextEdits
-		}, id);
+		};
+
+		if (item.insertText !== resolvedItem.insertText) {
```

Comment:
@jrieken thank you to revisit the issue.

To clarify, we did not change the `insertText` property when we resolved the completion item in the Java extension. Instead, we lazily resolved the `textEdit` property when the `completion/resolve` request was made. This is because the `insertText` value has no effect when a completion item has a `textEdit` property.

Question 1: Will this change cause the `textEdit` object from the resolvedItem to be ignored?

Question 2: Is this change included in the latest insider version (1.79.0-insider 2023-05-25T15:44:02.029Z)? I tested the latest insider on 2023-05-25 and the Java code completion still worked fine.

Reply:
I will check with @dbaeumer but I believe LSP maps its text edit always onto our API insert text and range

Reply:
@jrieken, yes that is what I do.

Reply:
@jrieken @dbaeumer got it, thanks for the confirmation.

Reply:
@testforstephen can you please double-confirm that Java is working fine (as before) with latest insiders. Thanks

Reply:
@jrieken Sure, I tested with the latest insider on 2023-06-01, and the lazy resolve behavior for Java extension works fine as before.